### ISSUE TITLE
Auth.currentCredentials and Auth.currentSession returns nothing on startup for 0.2.10 onward

### ISSUE THREAD

**Issue by xdl** [aws/aws-amplify#483]
In version 0.2.9, we were able to check the user's login status by calling `Auth.currentCredentials()` and checking if they have `expired`. If they have, we can redirect the user into the login page, using something like:

```
constructor() {
    Amplify.Auth.currentCredentials().then((result) => {
      if (result && !result.expired) {
          this.authState = "SignedIn";
        });
      } else {
        this.authState = "SignIn";
      }
  })
}
```

After upgrading to 0.2.10 and 0.2.11, the callback value for this (and `currentSession`) no longer return anything. However, later calls return the appropriate value, suggesting that there is some sort of timing issue involved.

Is this a regression, or the intended behaviour? If it is the latter, what would be the suggested way of checking whether the user's logged on when they first land on the app?

**Comment by Larsrdev** [comment]
Same here. LogIn is working. I get an CognitoUser back. But `Auth.currentSession()` and `Auth.currentCredentials()` says "No current user".

**Comment by powerful23** [comment]
@xdl Can you turn on the debug mode by putting ```window.LOG_LEVEL = 'DEBUG'``` to see if we could have more details. Also could you try calling this method in  ```componentDidMount``` to see if it works. In my opinion it's not a good behavior to call this method in ```constructor``` as you said maybe caused by some timing issue.

**Comment by powerful23** [comment]
@Larsrdev How did you call these methods?

**Comment by Larsrdev** [comment]
OnClickLogin via Button -> 
```
signIn(email: string, password: string): Promise<AuthEntity> {
        Amplify.Logger.LOG_LEVEL = 'DEBUG';
        return new Promise<AuthEntity>((resolve, reject) => {
            Auth.signIn(email, password)
                .then(user => {
                    resolve(new AuthEntity());
                }).catch(err => {
                reject(new ErrorEntity(err.toString(), -1));
            });
        });
    }
```
credentials are fetched and showed via log. I get a valid CognitoAccessToken, CognitoIdToken and CognitoRefreshToken.

But calling:
```
isLoggedIn(): Promise<AuthEntity> {
        return new Promise<AuthEntity>((resolve, reject) => {
            Auth.currentSession().then((session) => {
                resolve(new AuthEntity());
            }).catch((error) => {
                reject(new ErrorEntity(error.toString(), -1));
            });
        });
    }
```
rejects. 
And accessing credentials:
```
Auth.currentCredentials()
                .then(credentials => {
                    resolve(apigClientFactory.newClient({
                        accessKey: credentials.accessKeyId,
                        secretKey: credentials.secretAccessKey,
                        sessionToken: credentials.sessionToken,
                        region: process.env.REACT_APP_AWS_REGION,
                        invokeUrl: process.env.REACT_APP_AWS_HOST
                    }));
                });
```
Returns `Unhandled Rejection (TypeError): Cannot read property 'accessKeyId' of null`

I also got this INFO: `[INFO] 59:16.438 Cache - Get item: key is federatedInfo with options undefined`

aws-amplify@0.2.9 works. But aws-amplify@0.2.11 doesn't.

**Comment by powerful23** [comment]
@Larsrdev Can you give a try on aws-amplify@beta to see if it works for you? From the debug log I think the problem is that the credentials is not loaded when you call this method which should be fixed in the beta version.

**Comment by Larsrdev** [comment]
seems to be fixed in 0.2.12. 👍

**Comment by powerful23** [comment]
Closing the issue as this should have been fixed in the latest version.
